packages/camera/camera_web/lib/src/camera_service.dart

```diff
@@ -363,4 +364,78 @@ class CameraService {
         return DeviceOrientation.portraitUp;
     }
   }
+
+  ///Used to check if browser has OffscreenCanvas capability
+  bool hasPropertyOffScreenCanvas() {
+    return jsUtil.hasProperty(window, 'OffscreenCanvas'.toJS);
+  }
+
+  ///Used in [takeFrame] if [canUseOffscreenCanvas] is false
+  web.CanvasElement? _canvasElement;
+
+  ///Used in [takeFrame] if [canUseOffscreenCanvas] is false
+  web.OffscreenCanvas? _offscreenCanvas;
+
+  ///Returns frame at a specific time using video element
+  CameraImageData takeFrame(
+    web.VideoElement videoElement, {
+    bool canUseOffscreenCanvas = false,
+  }) {
+    final int width = videoElement.videoWidth;
+    final int height = videoElement.videoHeight;
+    if (width == 0 || height == 0) {
+      throw Exception(
+        'Computed dimensions are zero: width=$width, height=$height',
+      );
+    }
+    late web.ImageData imageData;
+    if (canUseOffscreenCanvas) {
+      if (_offscreenCanvas == null ||
+          _offscreenCanvas!.width != width ||
+          _offscreenCanvas!.height != height) {
+        _offscreenCanvas = web.OffscreenCanvas(width, height);
+      }
+      final web.OffscreenCanvasRenderingContext2D context =
+          _offscreenCanvas!.getContext(
+        '2d',
+        <String, dynamic>{'willReadFrequently': true}.toJSBox,
+      )! as web.OffscreenCanvasRenderingContext2D;
+      context.drawImage(videoElement, 0, 0);
+      imageData = context.getImageData(0, 0, width, height);
+    } else {
+      if (_canvasElement == null ||
+          _canvasElement!.width != width ||
+          _canvasElement!.height != height) {
+        _canvasElement = web.CanvasElement()
+          ..height = height
+          ..width = width;
+      }
+      final web.CanvasRenderingContext2D context = _canvasElement!.context2D;
+
+      context.drawImageScaled(
+        videoElement,
+        0,
+        0,
+        width.toDouble(),
+        height.toDouble(),
+      );
+      imageData = context.getImageData(0, 0, width, height);
+    }
+    final ByteBuffer byteBuffer = imageData.data.toDart.buffer;
+
+    return CameraImageData(
+      format: const CameraImageFormat(
+        ImageFormatGroup.jpeg,
+        raw: '',
+      ),
+      planes: <CameraImagePlane>[
+        CameraImagePlane(
+          bytes: byteBuffer.asUint8List(),
+          bytesPerRow: 0,
+        ),
+      ],
```

Comment:
Is this `CameraImageData` correctly constructed?

First it says that this is a `jpeg` (which it isn't), and then puts it in the first plane, but we say it's "0 bytesPerRow".

My suggestions are:

* Make the format `ImageFormatGroup.unknown` (it's RGB with 4 bytes per-pixel, arranged in rows, but that's not an expected ImageFormatGroup for the plugin).
* set a `raw` value that [doesn't clash with anything possible with Android or iOS](https://github.com/flutter/packages/blob/e55be9676ec6d519598dc716c5ec8c6dcb0e7973/packages/camera/camera_platform_interface/lib/src/types/camera_image_data.dart#L57-L66) .
* The `bytesPerRow` of the browser's `getImageData` seems to be `width * 4`. [Docs](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas#the_imagedata_object).

Reply:
I found `FLEX_RGBA_8888` in Android and `kCVPixelFormatType_32RGBA` in iOS which looks promising for compatibility with RGBA 4 bytes per-pixel arranged image coming from `ImageData` of Canvas (Although I haven't tested it yet)

Also found a comment in `\camera\camera_android_camerax\lib\src\android_camera_camerax.dart`
```dart
  /// [imageFormatGroup] is used to specify the image format used for image
  /// streaming, but CameraX currently only supports YUV_420_888 (supported by
  /// Flutter) and RGBA (not supported by Flutter). CameraX uses YUV_420_888
  /// by default, so [imageFormatGroup] is not used.
``` 

Seeing this, I went with the values you suggested

Reply:
I have tested this in the example app of [google_mediapipe_face_detection](https://github.com/TecHaxter/google_mediapipe_face_detection) (code not pushed yet). It works like charm ✨✨

Can we mark this resolved?